checks.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+# As described in `CONTRIBUTING.md`, this script checks Optuna's source codes for formatting,
+# coding style and type hints. The `-u` option enables automatic formatting by `black` and `isort`.
+
+res_pip_list=$(pip freeze)
+missing_dependencies=()
+if [ ! "$(echo $res_pip_list | grep black)" ] ; then
+  missing_dependencies+=(black)
+fi
+if [ ! "$(echo $res_pip_list | grep flake8)" ] ; then
+  missing_dependencies+=(hacking)
+fi
+if [ ! "$(echo $res_pip_list | grep isort)" ] ; then
+  missing_dependencies+=(isort)
+fi
+if [ ! "$(echo $res_pip_list | grep mypy)" ] ; then
+  missing_dependencies+=(mypy)
+fi
+if [ ! ${#missing_dependencies[@]} -eq 0 ]; then
+  echo "The following dependencies are missing:" "${missing_dependencies[@]}"
+  read -p "Would you like to install the missing dependencies? (y/N): " yn
+  case "$yn" in [yY]*) ;; *) echo "abort." ; exit ;; esac
+  pip install "${missing_dependencies[@]}"
+This conversation was marked as resolved by hvy
```

Comment:
A nit, but this line should be removed from the script as I encountered the following error:
```console
checks.sh: line 24: This: command not found
```

```suggestion
```